packages/protocol-parser/src/encodeRecord.ts

```diff
@@ -20,9 +20,12 @@ export function encodeRecord(schema: Schema, values: readonly (StaticPrimitiveTy
 
   const dynamicData = dynamicDataItems.join("");
 
-  const packedCounter = `${encodeField("uint56", dynamicTotalByteLength).replace(/^0x/, "")}${dynamicFieldByteLengths
-    .map((length) => encodeField("uint40", length).replace(/^0x/, ""))
-    .join("")}`.padEnd(64, "0");
+  const packedCounter =
+    schema.dynamicFields.length > 0
+      ? `${encodeField("uint56", dynamicTotalByteLength).replace(/^0x/, "")}${dynamicFieldByteLengths
+          .map((length) => encodeField("uint40", length).replace(/^0x/, ""))
+          .join("")}`.padEnd(64, "0")
+      : "";
```

Comment:
sorry, one more thing

it might make sense to do this check and return early so we avoid all the other work above

shortly after `const staticData = ...`, do something like
```ts
if (!dynamicValues.length) {
  return `0x${staticData}`;
}
```

Reply:
agree, this also mirrors the on-chain logic better